doc/level1.rst

```diff
@@ -637,29 +639,45 @@ Key Gossip
 ++++++++++
 
 It is a common use case to send an encrypted mail to a group of
-recipients. To ensure that these recipients can encrypt messages when
-replying to that same group, the keys of all recipients can be
-included in the encrypted payload. This does not include BCC
+recipients. To help these recipients to send encrypted replies
+to the same group, the keys of all recipients can be
+included in the encrypted payload through adding ``Autocrypt-Gossip``
+headers for each recipient . This does not include BCC
 recipients, which by definition must not be revealed to other
 recipients.
 
-The ``Autocrypt-Gossip`` header has the format as the ``Autocrypt``
+The ``Autocrypt-Gossip`` header has the same format as the ``Autocrypt``
 header (see `autocryptheaderformat`_). Its ``addr`` attribute
 indicates the recipient address this header is valid for as usual, but
 may relate to any recipient in the ``To`` or ``Cc`` header.
-See example in :ref:`autocrypt-gossip-example`
+See example in :ref:`autocrypt-gossip-example`.
+
+.. _`gossip-injection`:
 
 Key Gossip Injection in Outbound Mail
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-An Autocrypt MUA MAY include ``Autocrypt-Gossip`` headers in messages
+A MUA MAY include ``Autocrypt-Gossip`` headers in messages
 with more than one recipient. These headers MUST be placed in the root
 MIME part of the encrypted message payload. The encrypted payload in
-this case contains one Autocrypt-Gossip header for each recipient,
-which MUST include ``addr`` and ``keydata`` attributes with the
-corresponding values for the recipient identified by ``gossip-addr``
-as stored in ``peers[gossip-addr]``.  It SHOULD NOT contain a
-``prefer-encrypt`` attribute.
+this case contains one ``Autocrypt-Gossip`` header for each recipient,
+each of which MUST contain the ``addr``, ``keydata`` and the
+gossip-header specific ``origin-timestamp`` attributes. It SHOULD
+NOT contain the ``prefer-encrypt`` attribute.
+
+A MUA MUST gossip the same public key for a recipient that
+it uses to encrypt to that recipient. The values for ``keydata`` and
+``origin-timestamp`` are thus computed mirroring the key selection
+logic used in `preliminary-recommendation`_:
+
+- If ``peer-state[gossip-addr]`` contains ``null`` in its
+  ``public_key`` attribute, or its ``autocrypt_timestamp`` is
+  more than 35 days older than its ``gossip_timestamp``, set
+  ``keydata`` to ``gossip_key`` and ``origin_timestamp``
+  to ``gossip_timestamp``.
+
+- Otherwise, set ``keydata`` to ``public_key`` and
+  set ``gossip_timestamp`` to ``autocrypt_timestamp``.
```

Comment:
This puts Recommendation logic in the Gossip Injection section.  I think if we're going to modify or augment the recommendation, it should happen in the section that talks about the recommendation.

maybe the output of the recommendation algorithm needs to be augmented?